docs/DeveloperGuide.md

```diff
@@ -87,6 +123,34 @@ This section contains links to other relevant guides.
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Add task feature
+The adding of task mechanism is facilitated by `LogicManager`.
+
+The retrieving of task details is done by `Parser`. It splits the user input based on command type and the various parameters' description regardless of the order the user input the parameters in.
+The `AddCommand` class will then be executed and the task will be added to the `TaskList`.
+
+It implements the following operations:
```

Comment:
I think it is not clear what `It` refers to. How about mentioning `Parser`, `AddCommand` and `TaskList`.